pkg/instance/resources.go

```diff
@@ -0,0 +1,157 @@
+package instance
+
+import (
+	"context"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type resources struct {
+	instance      *Instance
+	policyRules   []rbacv1.PolicyRule
+	memoryRequest resource.Quantity
+	memoryLimit   resource.Quantity
+	cpuRequest    resource.Quantity
+}
+
+func (i *Instance) Resources() *resources {
+	return i.resources
+}
+
+// AddPolicyRule adds a policy rule to the instance
+// This function can only be called in the states 'Preparing' and 'Committed'
+func (r *resources) AddPolicyRule(rule rbacv1.PolicyRule) error {
+	if !r.instance.IsInState(StatePreparing, StateCommitted) {
+		return ErrAddingPolicyRuleNotAllowed.WithParams(r.instance.state.String())
+	}
+	r.policyRules = append(r.policyRules, rule)
+	return nil
+}
```

Comment:
Is there a reason this in in `resources` and not in `security`?

Reply:
There is no reason for the group names of the functions. I just put them in the groups that I thought it might make sense. Since you guys have more experience with `k8s`, please feel free to move the funcs in the groups that are more relevant or just comment on them and I will move them.

Reply:
Okay, I think this function makes more sense in `security`.